contracts/contracts/trade/src/contract.rs

```diff
@@ -676,49 +626,266 @@ fn refund_escrow(
+    let maker = offer.owner.clone();
+    let taker = if trade.seller.eq(&maker) {
+        trade.buyer.clone()
     } else {
+        trade.seller.clone()
+    };
+
+    // Check if winner is eligible, it must be either maker or taker
+    if winner.eq(&maker) {
+        trade.state = TradeState::SettledForMaker;
+    } else if winner.eq(&taker) {
         trade.state = TradeState::SettledForTaker;
+    } else {
+        return Err(ContractError::InvalidSender {
+            sender: winner,
+            buyer: trade.buyer,
+            seller: trade.seller,
+        });
     }
+    TradeModel::store(deps.storage, &trade).unwrap();
+
     // Pay arbitration fee
-    if arbitration_mode {
-        let fee_rate: Uint128 = Uint128::new(10);
-        let fee_amount = amount.multiply_ratio(Uint128::new(1), fee_rate);
+    let amount = trade.amount.clone();
+    let fee_rate: Uint128 = Uint128::new(10);
+    let fee_amount = amount.multiply_ratio(Uint128::new(1), fee_rate);
 
-        let fee = vec![Coin::new(fee_amount.u128(), denom.clone())];
-        let seller_amount = vec![Coin::new(amount.sub(fee_amount).u128(), denom.clone())];
+    let denom = denom_to_string(&trade.denom);
+    let fee = vec![Coin::new(fee_amount.u128(), denom.clone())];
+    let winner_amount = vec![Coin::new(
+        amount.u128().sub(fee_amount.u128()),
+        denom.clone(),
+    )];
 
-        let seller_msg = create_send_msg(trade.seller, seller_amount);
-        let arbitrator_msg = create_send_msg(trade.arbitrator.clone().unwrap(), fee);
+    let winner_msg = create_send_msg(winner.clone(), winner_amount);
+    let arbitrator_msg = create_send_msg(trade.arbitrator.clone().unwrap(), fee);
 
-        let res = Response::new()
-            .add_submessage(SubMsg::new(seller_msg))
-            .add_submessage(SubMsg::new(arbitrator_msg));
-        Ok(res)
-    }
-    //Arbitration Fee
-    if arbitration_mode {
-        let arbitration_amount = get_fee_amount(trade.amount.clone(), ARBITRATOR_FEE);
-        release_amount -= arbitration_amount;
-
-        // Send arbitration fee share
-        send_msgs.push(SubMsg::new(CosmosMsg::Bank(BankMsg::Send {
-            to_address: arbitrator.to_string(),
-            amount: vec![],
-        })));
-    }
-    */
+    let res = Response::new()
+        .add_attribute("arbitrator", trade.arbitrator.unwrap().to_string())
+        .add_attribute("winner", winner.to_string())
+        .add_attribute("maker", maker.to_string())
+        .add_attribute("taker", taker.to_string())
+        .add_submessage(SubMsg::new(winner_msg))
+        .add_submessage(SubMsg::new(arbitrator_msg));
+    Ok(res)
+}
+
+//region arbitration queries
+pub fn query_arbitrator(deps: Deps, arbitrator: Addr) -> StdResult<Vec<Arbitrator>> {
+    let storage = deps.storage;
+
+    let result = arbitrators()
+        .idx
+        .arbitrator
+        .prefix(arbitrator)
+        .range(storage, None, None, Order::Descending)
+        .take(10)
```

Comment:
Question: 
Why are we returning only 10 items in this query? Shouldn't it be a pagination parameter or return everything if we think it doesn't need pagination?

Reply:
Agreed, we can remove the limit for now.

Reply:
Addressed in https://github.com/Local-Money/localmoney/pull/80/commits/c3247b03c182cac9116322c32ffd594eff599898